python/cudf/cudf/io/json.py

```diff
@@ -21,13 +21,27 @@ def read_json(
     lines=False,
     compression="infer",
     byte_range=None,
+    keep_quotes=False,
     *args,
     **kwargs,
 ):
     """{docstring}"""
 
+    if not isinstance(dtype, (abc.Mapping, bool)):
+        raise TypeError("hi")
+        warnings.warn(
+            "passing 'dtype' as list is deprecated, "
+            "in future versions 'dtype' can only be a dict or bool",
+            FutureWarning,
+        )
+
     if engine == "cudf" and not lines:
         raise ValueError("cudf engine only supports JSON Lines format")
+    if engine == "cudf_experiment" and keep_quotes:
+        raise ValueError(
+            "keep_quotes='True' is supported only with"
+            " engine='cudf_experimental'"
+        )
```

Comment:
I think this check already covers it all no? i.e., this check will error on all other un-supported json reader `engines` when `keep_quotes=True`

Reply:
Ah, I explained myself badly. Right now you can pass `engine="cudf", orient=...` but most non-default `orient` options are not supported by the cudf engine IIUC.

I'll report some bugs because there are combinations that result in deadlocks and silently incorrect behaviour, but nothing else to do here.

Reply:
Ah okay, got it. Thanks for clarifying 👍